js/permissions.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+import * as msg from "gen/msg_generated";
+import * as flatbuffers from "./flatbuffers";
+import * as dispatch from "./dispatch";
+import { assert } from "./util";
+
+// Permissions as granted by the caller
+export type Permissions = {
+  read: boolean;
+  write: boolean;
+  net: boolean;
+  env: boolean;
+  run: boolean;
+
+  // NOTE: Keep in sync with src/permissions.rs
+};
+
+export type Permission = keyof Permissions;
+
+// Inspect granted permissions for the current program.
+export function permissions(): Permissions {
```

Comment:
Use jsdoc for the comment. Bonus points for a code example (ideally a working one that people can copy and paste)

Reply:
done